[FEAT] HomeView UI 구현 (#11)

kms0233 opened the pull request:
# 🔥 *Pull requests*

## ⛳️ **작업한 브랜치**
- feature/#11

## 👷 **작업한 내용**
<!-- 작업한 내용을 적어주세요. -->
HomeView Compositional Layout으로 UI 구현했습니다!

## 🚨 참고 사항
<!-- 참고할 사항이 있다면 적어주세요. -->
- **무한 스크롤 + 바텀 페이지 인디케이터**
4시간 삽질했는데 실패해서 .. 추후  따로 이슈 파서 구현하도록 하겠습니다!
- **추천하는 공고 Cell**
공고 Cell 재현이가 구현해줘서 머지하면 나중에 수정하도록 할게요! 일단은 게시판 Cell로 임시로 넣어두었습니다!
###  1️⃣ TagHeaderView
공통으로 사용하게 될 TagHeaderView 구현했습니다
Model 형식에 맞춰서 더미 넣은 후 사용하면 됩니다~
```
struct TagHeader {
    let nickname: String
    let title: String
    let discription: String
    let tags: [String]
}
```
### 2️⃣ CategorySelectorCell
거의 모든 뷰에서 공통으로 사용되는 카테고리 버튼이 있어서 공통 Cell 구현해두었습니다!

`configure(with: isSelected:)`는 셀에 카테고리 데이터를 바인딩하고,
선택 상태를 기반으로 스타일을 적용하는 메서드를 호출합니다!
`updateSelectionState(isSelectedCell:)`는 셀이 선택되었는지 상태에 따라 배경색과 텍스트 색상을 설정해줍니당

```
extension CategorySelectorCell {
    
    func configure(with category: CategorySelector, isSelected: Bool) {
        categoryLabel.text = category.title
        updateSelectionState(isSelectedCell: isSelected)
    }
    
    func updateSelectionState(isSelectedCell: Bool) {
        if isSelectedCell {
            self.backgroundColor = .lkBlue
            categoryLabel.textColor = .lkWhite
        } else {
            self.backgroundColor = .gray300 
            categoryLabel.textColor = .gray900
        }
    }

}
```

### 3️⃣ PolicyFooterView
홈 뷰에서 쓰이는 PolicyFooterView 구현해두었습니당

## ‼️ 논의 사항
- 스크롤되는 VC일때, 하단 탭바의 backgroundColor가 바뀌는 현상있어서 확인부탁드립니다! 
- 여러 뷰에서 공통을 쓰이는 Cell들이 많은데 이거 다 UICoponent에 폴더링하면 될까용??


## 📸 스크린샷
|기능|스크린샷|
|:--:|:--:|
|아이폰15Pro|![Simulator Screen Recording - iPhone 13 mini - 2024-11-21 at 17 58 16](https://github.com/user-attachments/assets/9c1840e8-946c-4792-b933-a5ef54d1e762)|

## 💫 관련 이슈
- Resolved: #11

kms0233 committed:
[Chore] 스크랩 아이콘 추가 (#11)

kms0233 committed:
[Feat] Home banner cell UI 구현 (#11)

kms0233 committed:
[Feat] Board cell UI 구현 (#11)

kms0233 committed:
[Feat] TagHeaderView UI 구현 (#11)

kms0233 committed:
[Feat] HomeModel 구현 (#11)

kms0233 committed:
[Feat] HomeSection enum 구현 (#11)

kms0233 committed:
[Feat] HomeView Compositional layout 구현 (#11)

kms0233 committed:
[Chore] BottomPageControlView 생성 (#11)

kms0233 committed:
[Feat] Compositional Layout cell 등록 (#11)

kms0233 committed:
[Add] 더미 데이터 추가 (#11)

kms0233 committed:
[Feat] Cell 데이터 바인딩 구현 (#11)

kms0233 committed:
[Feat] TagHeaderView 데이터 바인딩 구현 (#11)

kms0233 committed:
Merge branch 'main' into feature/#11

kms0233 committed:
[Feat] PolicyFooterView UI 구현 (#11)

kms0233 committed:
[Feat] PolicyFooterView 레이아웃 구현 (#11)

kms0233 committed:
[Feat] PolicyFooterView register (#11)

kms0233 committed:
[Chore] setLabel extension으로 수정 (#11)

kms0233 committed:
[Fix] HomeModel 데이터 바인딩 수정 (#11)

kms0233 committed:
[Fix] UILabel 레이아웃 수정 (#11)

kms0233 committed:
[Feat] Category Button Cell UI 구현 (#11)

kms0233 committed:
[Feat] Category Selector Section 구현 (#11)

kms0233 committed:
[Del] 필요없는 공백 & 주석 삭제 (#11)

Ohjackson commented on LINKAREER-iOS/LINKAREER-iOS/Presentation/Home/View/Cell/BoardCell.swift:
```diff
@@ -0,0 +1,190 @@
+
+extension BoardCell {
+    
+    func configure(with board: Board) {
```
good job~!

mcrkgus commented on LINKAREER-iOS/LINKAREER-iOS/Presentation/Home/View/HomeView.swift:
```diff
@@ -0,0 +1,169 @@
+    
+    
+    func setHierarchy() {
+        self.addSubviews(mainCollectionView)
```
addSubview로 바꿔주세여

mcrkgus commented on LINKAREER-iOS/LINKAREER-iOS/Presentation/Home/View/PolicyFooterView.swift:
```diff
@@ -0,0 +1,92 @@
+import SnapKit
+import Then
+
+class PolicyFooterView: UICollectionReusableView {
```
final 키워드 빠졌어용

Ohjackson commented on LINKAREER-iOS/LINKAREER-iOS/Presentation/Home/Model/HomeModel.swift:
```diff
@@ -0,0 +1,79 @@
+    }
+}
+
+struct TagHeader {
```
많이 배워 갑니다~!

mcrkgus commented on LINKAREER-iOS/LINKAREER-iOS/Presentation/Home/View/TagHeaderView.swift:
```diff
@@ -0,0 +1,134 @@
+                $0.contentEdgeInsets = UIEdgeInsets(top: 2, left: 6, bottom: 2, right: 6)
+            }
+            
+            tagButton.snp.makeConstraints {
+                $0.height.equalTo(17)
+                $0.width.greaterThanOrEqualTo(50)
+            }
+            
+            tagStackView.addArrangedSubview(tagButton)
```
이거 레이아웃 먼저 잡아줘도 순서는 상관없나요??

kms0233 replied:
흠 동작은 잘되지만 레이아웃 충돌 있을 수 있으니 수정하도록 하겠습니닷!

mcrkgus commented on LINKAREER-iOS/LINKAREER-iOS/Presentation/Home/ViewController/HomeViewController.swift:
```diff
@@ -0,0 +1,139 @@
+//
+
+import UIKit
+
```
라이브러리 빠졌서용

mcrkgus reviewed (approved):
수고하셨습니당 컴포지셔널 굿

Ohjackson reviewed (approved):
고생하셨습니다~!! LGTM~~

kms0233 committed:
[Fix] 코드리뷰 반영 (#11)

kms0233 committed:
[Chore] 공통 컴포넌트 UIComponent로 이동 (#11)

kms0233 committed:
[Merge] branch 'main' into feature (#11)

# Conflicts:
#	LINKAREER-iOS/LINKAREER-iOS/Presentation/TabBar/TabBarItem.swift
